app.js

```diff
@@ -57,16 +78,17 @@ function onVote(event) {
   event.preventDefault();
   var productInput = [];
   for (var j = 0; j < 3; j++) {
-    productInput[j] = event.target.querySelector('input[value="' + j + '"]');
+    productInput[j] = event.target.querySelector('input[id="' + currentProducts[j].name + '"]');
```

Comment:
This query is going to fail on products that have a space in their `name`, as the one restriction on `id` values in HTML5 is that they [must not contain spaces](https://www.w3.org/TR/html5/dom.html#the-id-attribute).